dashboard/app/controllers/projects_lists_controller.rb

```diff
@@ -0,0 +1,41 @@
+class ProjectsListsController < ApplicationController
+  include LevelsHelper
+
+  def index
+    section_id = params[:section_id]
+    section_owner_id = Section.find(section_id).user_id
+    raise "section can only be accessed by its owner" unless current_user && current_user.id == section_owner_id
+
+    @projects_list_data = []
+    Section.find(section_id).students.each do |student|
+      student_storage_id = PEGASUS_DB[:user_storage_ids].where(user_id: student.id).first
+      next unless student_storage_id
+      PEGASUS_DB[:storage_apps].where(storage_id: student_storage_id[:id], state: 'active').each do |project|
+        # The channel id stored in the project's value field may not be reliable
+        # when apps are remixed, so recompute the channel id.
+        channel_id = storage_encrypt_channel_id(student_storage_id[:id], project[:id])
+        project_row_data = get_project_row_data(student, project, channel_id)
+        @projects_list_data << project_row_data if project_row_data
+      end
+    end
+  end
+
+  private
+
+  # e.g. '/p/applab' -> 'applab'
+  def project_type(level)
+    level && level.split('/')[2]
+  end
+
+  def get_project_row_data(student, project, channel_id)
+    project_value = project[:value] ? JSON.parse(project[:value]) : {}
+    return nil if project_value['hidden'] == true || project_value['hidden'] == 'true'
+    {
+      channel: channel_id,
+      name: project_value['name'],
+      studentName: student.name,
+      type: project_type(project_value['level']),
+      updatedAt: project_value['updatedAt'],
+    }
+  end
```

Comment:
No, this logic is not duplicated anywhere else, but I'm fine with moving it. Please advise on whether I should move it to a helper or a model.

a few thoughts:
* since we're not persisting project lists to the db, we might not need a model
* using a model, I could to pull in some values "through" other models such as users/students and sections rather that using explicit ActiveRecord calls. caveat: I won't be able to use activerecord to access `user_storage_ids` or `storage_apps` / projects since those are in pegasus.